### PR TITLE
Resolving relative application base paths

### PR DESCRIPTION
**Pull request by moozzyk**
This allows providing relative paths in ASPNET_APPLICATIONBASE variable

### CODE REVIEW

File: src/Microsoft.AspNetCore.Hosting/WebHostBuilder.cs [hunk 1]
```diff
@@ -224,7 +225,7 @@ private class WrappedApplicationEnvironment : IApplicationEnvironment
         {
             public WrappedApplicationEnvironment(string applicationBasePath, IApplicationEnvironment env)
             {
-                ApplicationBasePath = applicationBasePath;
+                ApplicationBasePath = Path.GetFullPath(applicationBasePath);
```
Comment (Tratcher) [inline src/Microsoft.AspNetCore.Hosting/WebHostBuilder.cs]: This resolves relative to CurrentDirectory, which is very arbitrary, no? I thought you were going to resolve relative to the exe path?

Comment (moozzyk) [inline src/Microsoft.AspNetCore.Hosting/WebHostBuilder.cs]: When we talked the idea was to resolve relative to the .exe but for a deployed app isn't it more useful/natural to resolve it relative to a current directory?

Comment (davidfowl) [inline src/Microsoft.AspNetCore.Hosting/WebHostBuilder.cs]: No, it should be resolved based on the exe directory

Comment (moozzyk) [inline src/Microsoft.AspNetCore.Hosting/WebHostBuilder.cs]: OK

**Comment by moozzyk** [comment]
:up::date:

**Comment by moozzyk** [comment]
:watch:

File: src/Microsoft.AspNetCore.Hosting/WebHostBuilder.cs [hunk 2]
```diff
@@ -238,5 +239,22 @@ public WrappedApplicationEnvironment(string applicationBasePath, IApplicationEnv
...
+            }
+
+            var basePath =
+#if NET451
+                (string)AppDomain.CurrentDomain.GetData("APP_CONTEXT_BASE_DIRECTORY") ??
+                AppDomain.CurrentDomain.BaseDirectory;
+#else
+                AppContext.BaseDirectory;
+#endif
+            return Path.Combine(Path.GetFullPath(basePath), applicationBasePath);
```
Comment (Tratcher) [inline src/Microsoft.AspNetCore.Hosting/WebHostBuilder.cs]: In which cases is basePath relative such that you need GetFullPath?

Comment (moozzyk) [inline src/Microsoft.AspNetCore.Hosting/WebHostBuilder.cs]: Fair point. As per our conversation I will do GetFullPath on the combined result to collapse unneeded traversals.

**Pull request merged**